docs/pages/tutorials/python.rst

```diff
@@ -1,3 +1,142 @@
+
+.. tip::
+
+    All of the examples scripts that follow can be found :gh_file:`here <examples/python>`.
+
+Mapping
+=======
+The only requirements to build wavemap maps are that you have a set of
+
+1. depth measurements,
+2. sensor pose (estimates) for each measurement.
+
+We usually use depth measurements from depth cameras or 3D LiDARs, but any source would work as long as a corresponding :ref:`projection <configuration_projection_models>` and :ref:`measurement <configuration_measurement_models>` model is available. To help you get started quickly, we provide example configs for various sensor setups :gh_file:`here <interfaces/ros1/wavemap_ros/config>`. An overview of all the available settings is provided on the :doc:`parameters page <../parameters/index>`.
+
+Example pipeline
+----------------
+
+.. literalinclude:: ../../../examples/python/mapping/full_pipeline.py
+    :language: python
+
+Serialization
+=============
+Next, we show how you can serialize and deserialize common wavemap objects, for example to save and load them from files.
+
+Maps
+----
+Wavemap uses a lightweight, efficient binary format to serialize its maps. The same format is used across wavemap's C++, Python and ROS interfaces. You could therefore, for example, create maps on a robot with ROS and subsequently analyze them in Python.
+
+Binary files
+^^^^^^^^^^^^
+Maps can be saved to disk using
+
+.. literalinclude:: ../../../examples/python/io/save_map_to_file.py
+    :language: python
+
+.. _python-code-examples-read-map:
+
+and read with
+
+.. literalinclude:: ../../../examples/python/io/load_map_from_file.py
+    :language: python
+
+Configs
+-------
+In the previous mapping pipeline example, the configuration parameters for the map and the measurement integration components were hard-coded. To make your setup more flexible, you can use configuration files. We will demonstrate how to work with YAML files, which is the format we use for wavemap's :gh_file:`example configs <interfaces/ros1/wavemap_ros/config>`. However, pywavemap is flexible and can support any parameter format that can be read into a Python `dict`.
+
+
+YAML files
+^^^^^^^^^^
+
+.. literalinclude:: ../../../examples/python/io/load_params_from_file.py
+    :language: python
+
+
+Queries
+=======
+In this section, we show how you can query wavemap maps and classify whether a point or region of interest is occupied.
+
+Node indices
+------------
+The map models the environment by filling it with cubes of variable sizes, arranged as the nodes of an octree. Node indices are defined as integer [X, Y, Z, height] coordinates, whose XYZ values correspond to the node's position in the octree's grid at the given *height*, or level in the tree. Height 0 corresponds to the map's maximum resolution, and the grid resolution is halved for each subsequent height level.
```

Comment:
Same comment as in the C++ section about the inclusion of an illustration of the concept.